libsolidity/interface/CompilerStack.cpp

```diff
@@ -1736,13 +1736,12 @@ std::string CompilerStack::createMetadata(Contract const& _contract, bool _forIR
 		}
 		else if (OptimiserSuite::isEmptyOptimizerSequence(m_optimiserSettings.yulOptimiserSteps + ":" + m_optimiserSettings.yulOptimiserCleanupSteps))
 		{
-			solAssert(m_optimiserSettings.optimizeStackAllocation == false);
 			details["yulDetails"] = Json::object();
+			details["yulDetails"]["stackAllocation"] = m_optimiserSettings.optimizeStackAllocation;
 			details["yulDetails"]["optimizerSteps"] = ":";
 		}
 		else
 		{
-			solAssert(m_optimiserSettings.optimizeStackAllocation == false);
 			solAssert(m_optimiserSettings.yulOptimiserSteps == OptimiserSettings::DefaultYulOptimiserSteps);
```

Comment:
Maybe we should add the `optimizeStackAllocation` setting to the metadata here as well. I didn’t include it because `yulDetails` is omitted when the `optimizer` is disabled. However, now it is also being included when an empty optimizer sequence is provided, which wasn’t the case previously.  

Since `optimizeStackAllocation` can be either `true` or `false` in this else block, I think we should include it to the metadata here. That said, this would essentially mean always including `yulDetails` in the metadata, and I’m not sure if that’s acceptable.